### ISSUE TITLE
Many rule allow unwanted additionalProperties

### ISSUE THREAD

**Issue by Dimava** [eslint-types/eslint-define-config#189]
I see over 25 rules with
```ts
[k: string]: any;
```
which disappear if `json-schema-to-typescript` is configured with `{ additionalProperties: /* default to: */ false }`

I think at least 75% of that is schema typos, and we should use `{ additionalProperties: /* default to: */ false }`

If the type actually has additionalProperties it is always written in type (as `Record` types obviously require them to exist), while when it hasn't it may be just be forgotten

**Comment by Dimava** [comment]
![image](https://user-images.githubusercontent.com/20068737/233735391-43506a1f-fd93-406f-b084-079d9eb0b4d2.png)

**Comment by Shinigami92** [comment]
Oh nice there is a property for that? wow? This could actually fix all our issues :tada:

**Comment by Shinigami92** [comment]
I will check that tomorrow or you already did a PR

**Comment by Dimava** [comment]
I didnt

**Comment by Shinigami92** [comment]
> I didnt

I meant: I will do a PR on my own tomorrow if you did not already send a PR from your side

sorry for misunderstandment :sweat_smile: 
I think english is not the native language for both of us :sweat_smile:

**Comment by Dimava** [comment]
Other things to fix are
```ts
        ignore?: (string & { // string:not(...):not(...), should be just string here
          [k: string]: any;
        } & {
          [k: string]: any;
        })[];
```
and 
```ts
        ignorePatterns?: any[]; // mostly string[]
```
and maybe a couple of annoying TS types that depend on ESLint types

**Comment by Dimava** [comment]
> > I didnt
> 
> I meant: I will do a PR on my own tomorrow if you did not already send a PR from your side
> 
> sorry for misunderstandment 😅 I think english is not the native language for both of us 😅

In my language "I will check that tomorrow or you already did a PR" would be read as a question, so I answered that 😅

**Comment by Dimava** [comment]
Some context-sensitive types like
```ts
export type regex = string & { _?: regex }; // string that is used as RegExp pattern
export type glob = string & { _?: regex }; // string which is used as glob pattern
```
could also be (very) useful, but that's hard to mantain

**Comment by Shinigami92** [comment]
I tried to configure `additionalProperties` to `true` and `false` but nothing changed

https://github.com/Shinigami92/eslint-define-config/blob/c8489871f1e409f306cd290051a09d4773ebb807/scripts/generate-rule-files/src/json-schema-to-ts.ts#L26-L34

So I think we will just use patching again
But we can apply this patch generally like we did here

https://github.com/Shinigami92/eslint-define-config/blob/c8489871f1e409f306cd290051a09d4773ebb807/scripts/generate-rule-files/src/json-schema-to-ts.ts#L8-L11

**Comment by Shinigami92** [comment]
@Dimava The `[k: string]: any;` itself is not a problem at all, it just means that other properties can be added to the object and mostly if you need that you should check the website documentation of the rule.

Only stuff like `string & otherStuff` is something like a bug and we might need to report that back to `json-schema-to-typescript` repo

But for now I fixed it with #190

**Comment by Shinigami92** [comment]
plugin types are now supported via @eslint-type/*
if there are any `additionalProperties` I think they need to be fixed in their respective meta schemas in the specific plugin
